samples/bluetooth/mesh/sensor_client/src/model_handler.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/mesh/models.h>
+#include <dk_buttons_and_leds.h>
+#include "model_handler.h"
+
+#define GET_DATA_TIMER_INTERVAL 3
+
+static void sensor_cli_data_cb(struct bt_mesh_sensor_cli *cli,
+			       struct bt_mesh_msg_ctx *ctx,
+			       const struct bt_mesh_sensor_type *sensor,
+			       const struct sensor_value *value)
+{
+	if (sensor->id == bt_mesh_sensor_present_dev_op_temp.id) {
+		printk("Chip temperature: %d.%06d;\n", value->val1,
+		       value->val2);
+	} else if (sensor->id == bt_mesh_sensor_presence_detected.id) {
+		printk("Precense detected\n");
+	} else if (sensor->id ==
+		   bt_mesh_sensor_time_since_presence_detected.id) {
+		if (value->val1) {
+			printk("%d second(s) since last presence detected\n",
+			       value->val1);
+		} else {
+			printk("No presence detected, or under 1 second since presence detected\n");
+		}
+	}
+}
+
+static void sensor_cli_series_entry_cb(
+	struct bt_mesh_sensor_cli *cli, struct bt_mesh_msg_ctx *ctx,
+	const struct bt_mesh_sensor_type *sensor, u8_t index, u8_t count,
+	const struct bt_mesh_sensor_series_entry *entry)
+{
+	printk("Relative runtime in %d to %d degree: %d.%d percent\n",
+	       entry->value[1].val1, entry->value[2].val1, entry->value[0].val1,
+	       entry->value[0].val2);
+}
+
+static const struct bt_mesh_sensor_cli_handlers bt_mesh_sensor_cli_handlers = {
+	.data = sensor_cli_data_cb,
+	.series_entry = sensor_cli_series_entry_cb,
+};
+
+static struct bt_mesh_sensor_cli sensor_cli =
+	BT_MESH_SENSOR_CLI_INIT(&bt_mesh_sensor_cli_handlers);
+
+static void get_data_timer_handler(struct k_timer *dummy)
+{
+	int err;
+
+	err = bt_mesh_sensor_cli_series_entries_get(
+		&sensor_cli, NULL,
+		&bt_mesh_sensor_rel_runtime_in_a_dev_op_temp_range, NULL, NULL,
+		NULL);
+	if (err) {
+		printk("Error getting relative chip temperature data (%d)\n",
+		       err);
+	}
+
+	err = bt_mesh_sensor_cli_get(
+		&sensor_cli, NULL, &bt_mesh_sensor_time_since_presence_detected,
+		NULL);
+	if (err) {
+		printk("Error getting time since presence detected (%d)\n",
+		       err);
+	}
+}
+
+K_TIMER_DEFINE(get_data_timer, get_data_timer_handler, NULL);
```

Comment:
Unfortunately, my parents refused to sign anything because I am "a grown man" that "can take care of my own", so i had to change from timer to delayed work